src/mds/objects.py

```diff
@@ -100,7 +115,7 @@ async def create_object(
     metadata = metadata or {}
 
     # get user id from token claims
-    uploader = token_claims.get("sub")
+    uploader = token_claims.get("context", {}).get("user", {}).get("name")
```

Comment:
is there a requirement to change this or was just to show the human readable username? the reason it's sub is b/c in some instances the username is controlled protected information and the metadata service does not have filtering for authorization at the moment, so anyone can get to this information

Reply:
i don't remember the exact flow, but i created a blank record in indexd, the uploader was the username (automatically). i tried to use the MDS object creation endpoint and got a 403 from indexd because the uploader did not match (sub VS username)

We do have a ticket in the backlog to use sub instead of the username in indexd (PXP-4871), but until that's done, i think they should match?

Reply:
blank record creation should not require an uploader anymore in indexd, were you using the latest version?

Reply:
i changed it back to sub. i think the issue might have been when creating a new version of an old blank record with an `uploader`